clients/drcachesim/tools/view.cpp

```diff
@@ -118,37 +146,26 @@ view_t::process_memref(const memref_t &memref)
         std::cerr << "T" << memref.instr.tid << " ";
     }
 
-    if (memref.marker.type == TRACE_TYPE_MARKER) {
-        if (memref.marker.tid != 0 &&
-            printed_header_.find(memref.marker.tid) == printed_header_.end()) {
-            printed_header_.insert(memref.marker.tid);
+    if (memref.marker.type == TRACE_TYPE_MARKER ||
+        (knob_sim_refs_ > 0 && instr_count_ == knob_skip_refs_)) {
```

Comment:
Why did we need this additional condition? Is it for the case when we don't have any markers in the [knob_skip_refs_, knob_skip_refs_ + knob_sim_refs_) instr range?

Reply:
Yes, exactly.  Adding a coment.